libraries/core_libs/network/src/tarcap/packets_handlers/transaction_packet_handler.cpp

```diff
@@ -106,8 +121,8 @@ void TransactionPacketHandler::onNewTransactions(
 }
 
 void TransactionPacketHandler::periodicSendTransactions(SharedTransactions &&transactions) {
-  std::unordered_map<dev::p2p::NodeID, std::vector<taraxa::bytes>> transactions_to_send;
-  std::unordered_map<dev::p2p::NodeID, std::vector<trx_hash_t>> transactions_hash_to_send;
+  std::unordered_map<dev::p2p::NodeID, std::pair<std::vector<trx_hash_t>, std::vector<taraxa::bytes>>>
+      transactions_to_send;
```

Comment:
we send the same transaction to multiple peers, if I would do it this way, would I not be serializing transaction to bytes for each peer so it would be done multiple times?

Reply:
good and valid point. But we are doing it even now if you check the code -> this `transactions_to_send[peer.first].second.push_back(trx->rlp());` is called for each peer

Reply:
actually I see now that we cache the rlp inside of transaction object so your approach is good afterall. I will change it

Reply:
Fixed